src/bin/compute-wer-confidence.cc

```diff
@@ -0,0 +1,247 @@
+    RandomAccessTokenVectorReader hyp_reader(hyp_rspecifier);
+    RandomAccessTokenVectorReader hyp_reader2(hyp_rspecifier2);
+    int32 num_words = 0, word_errs = 0, 
+            num_ins = 0, num_del = 0, num_sub = 0;
+    
+    // Main loop, store WER stats per hyp,
+    for (; !ref_reader.Done(); ref_reader.Next()) {
+      std::string key = ref_reader.Key();
+      const std::vector<std::string> &ref_sent = ref_reader.Value();
+      std::vector<std::string> hyp_sent, hyp_sent2;
+      if (!hyp_reader.HasKey(key) && !hyp_reader2.HasKey(key)) {
+          KALDI_ERR << "No hypothesis for key " << key << " in both transcripts "
+              "comparison is not possible.";
+      } else {
+        hyp_sent = hyp_reader.Value(key);
+        hyp_sent2 = hyp_reader2.Value(key);
+      }
+      num_words = ref_sent.size();
+
+      word_errs = LevenshteinEditDistance(ref_sent, hyp_sent, 
+                                            &num_ins, &num_del, &num_sub);
+      edit_word_per_hyp.push_back(std::pair<int32, int32>(word_errs, num_words));
+
+      word_errs = LevenshteinEditDistance(ref_sent, hyp_sent2, 
+                                            &num_ins, &num_del, &num_sub);
+      edit_word_per_hyp2.push_back(std::pair<int32, int32>(word_errs, num_words));
+    }
+
+}
+
+void GetBootstrapWERInterval(
+      const std::vector<std::pair<int32, int32> > & edit_word_per_hyp,
+      int32 replications, 
+      BaseFloat *mean, BaseFloat *interval) {
+    BaseFloat wer_accum = 0.0, wer_mult_accum = 0.0;
+
+    for (int32 i = 0; i <= replications; ++i) {
+      int32 num_words = 0, word_errs = 0;
+      for (int32 j = 0; j <= edit_word_per_hyp.size(); ++j) {
+        int32 random_pos = kaldi::RandInt(0, edit_word_per_hyp.size());
+        word_errs += edit_word_per_hyp[random_pos].first;
+        num_words += edit_word_per_hyp[random_pos].second;
+        }
+
+      BaseFloat wer_rep = static_cast<BaseFloat>(word_errs) / num_words;
+      wer_accum += wer_rep;
+      wer_mult_accum += wer_rep*wer_rep;
+    }
+
+    // Compute mean WER and std WER
+    *mean = wer_accum / (replications-1);
+    *interval = 1.96*sqrt(wer_mult_accum/(replications-1)-(*mean)*(*mean));
+}
+
+void GetBootstrapWERTwoSystemComparison(
+      const std::vector<std::pair<int32, int32> > & edit_word_per_hyp,
+      const std::vector<std::pair<int32, int32> > & edit_word_per_hyp2,
+      int32 replications, BaseFloat *p_improv) {
+    int32 improv_accum = 0.0;
+
+    for (int32 i = 0; i <= replications; ++i) {
+      int32 word_errs = 0;
+      for (int32 j = 0; j <= edit_word_per_hyp.size(); ++j) {
+        int32 random_pos = kaldi::RandInt(0, edit_word_per_hyp.size());
+        word_errs += edit_word_per_hyp[random_pos].first - 
+                        edit_word_per_hyp2[random_pos].first;
+        }
+      if(word_errs > 0)
+        ++improv_accum;
+    }
+    // Compute mean WER and std WER
+    *p_improv = static_cast<BaseFloat>(improv_accum) / (replications-1);
```

Comment:
This does not seem right -- shouldn't the p_improv be straight ratio between improv_accum and replication? I.e. not "replications-1"
Update: I looked into the paper and it seems the paper is indeed using "improv_accum/replications"